/*
Copyright 2025 Multiscale Modeling of Fluid Materials, TU Munich

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package chemsim;

message Model {

  required string mlir_module = 1;

  enum NeighborListType {
    SIMPLE_SPARSE = 0;
    SIMPLE_DENSE = 2;
    DEVICE_SPARSE = 1;
  }

  message NeighborList {
    required NeighborListType type = 1;
    required float cutoff = 2;

    // To which order neighbors are required for the newton and the
    // non-newton setting
    repeated int32 nbr_order = 3;

    optional bool half_list = 4; // Expect only a half neighbor list
  }

  required NeighborList neighbor_list=4;

  optional string unit_style = 5; // LAMMPS unit style definition

}
